iothub/service/src/Twin/Models/TwinCollection.cs

```diff
@@ -115,6 +115,7 @@ public int Count
         /// </summary>
         /// <param name="propertyName">Name of the property to get.</param>
         /// <returns>Value for the given property name.</returns>
+        /// <exception cref="InvalidOperationException">When the specified <paramref name="propertyName"/> does not exist in the collection.</exception>
         public dynamic this[string propertyName]
```

Comment:
We've chosen this as the exception of choice when the user does something the SDK doesn't support at runtime.